### PR TITLE
chore: Wire loggers in CKF correctly

### PR DESCRIPTION
**Pull request by andiwand**
That way updater and smoother logs will not be automatically silenced

### CODE REVIEW

File: Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp [hunk 1]
```diff
@@ -950,7 +950,7 @@ class CombinatorialKalmanFilter {
         } else {
           // Kalman update
           auto updateRes = m_extensions.updater(
-              gctx, trackState, Direction::Forward, getDummyLogger());
+              gctx, trackState, Direction::Forward, *logger().clone("updater"));
```
Comment (paulgessinger) [inline Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp:953]:
```suggestion
              gctx, trackState, Direction::Forward, *logger().cloneWithSuffix("Updater"));
```

File: Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp [hunk 2]
```diff
@@ -1129,9 +1129,9 @@ class CombinatorialKalmanFilter {
       ACTS_VERBOSE("Apply smoothing on " << nStates
                                          << " filtered track states.");
       // Smooth the track states
-      auto smoothRes =
-          m_extensions.smoother(state.geoContext, *result.fittedStates,
-                                lastMeasurementIndex, getDummyLogger());
+      auto smoothRes = m_extensions.smoother(
+          state.geoContext, *result.fittedStates, lastMeasurementIndex,
+          *logger().clone("smoother"));
```
Comment (paulgessinger) [inline Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp:1134]:
```suggestion
          *logger().cloneWithSuffix("Smoother"));
```

**Review by paulgessinger — COMMENTED** [review]
How about this?

Also, the logger should be created ahead of time, otherwise we do a lot of unnecessary heap allocations. Also consider that this is only not dangling because of lifetime extension (`clone()` returns a `unique_ptr`, which you dereference), which I'd prefer to avoid.

**Comment by andiwand** [comment]
good point - I adopted this now

**Comment by paulgessinger** [comment]
The job queue overflow fails sentinel, I'm merging manually.

**Pull request merged**